### DISCUSSION TITLE
Epiphany: The first day after Sunday Epiphany seems funny.

### DISCUSSION THREAD

**Discussion by pvanthony** [romcal/romcal#322] [General]
For version 3, here is our calendar setting.
```
// create object romcal
const romcal = new Romcal({
    localizedCalendar: GeneralRoman_En,
    scope: 'gregorian',
    epiphanyOnSunday: true,
    corpusChristiOnSunday: true,
    ascensionOnSunday: false,
    outputOptionalMemorials: false,
    strictMode: true
});
```
For us, Epiphany is on Sunday. The first day after Epiphany Sunday is 2022-01-03 but it shows "DAYS_BEFORE_EPIPHANY". Then on the second day after Epiphany is showing correctly, 2022-01-04, "DAYS_FROM_EPIPHANY".

This seems to be happening only for the first day after Epiphany. I am wondering if I have something wrong.

Need advice on how to setup romcal correctly.

**Comment by tukusejssirs** [comment]
I cannot test this ATM, but is seems to be a bug.

**Reply by tukusejssirs** [reply]
@emagnier, I can confirm this being a bug. Can we ~~move this to GitHub Issues~~ create an issue from this discussion?

**Reply by emagnier** [reply]
Yes, I'll try to fix to have a look and fix it in the coming next days.

**Reply by emagnier** [reply]
Issue created here: https://github.com/romcal/romcal/issues/327

**Reply by pvanthony** [reply]
Is there some code that I can put on my end as a temp fix? If yes, please share.

**Reply by emagnier** [reply]
Yes, you can check the `year` and the `key` of this specific day object, and adjust the name or anything else according to your need.

**Reply by pvanthony** [reply]
Ok. I will try that.

**Reply by pvanthony** [reply]
Tried to get a unique key for the Mondays after epiphany but the keys are not the same for every Monday after epiphany. Maybe I am doing something wrong. 
If there is time, checkout the console log at https://romcal.sound.sg/auto/index.html
Here is the code used to output the console log.
```
            for(day in result){
                if(result[day][0].periods[0]== "DAYS_BEFORE_EPIPHANY"){
                    console.log("Days Before Epiphany: ", result[day][0].date, result[day][0].name, result[day][0].colorNames)
                }
            }

            for(day in result){
                if(result[day][0].periods[0]== "DAYS_FROM_EPIPHANY"){
                    console.log("Days From Epiphany: ", result[day][0].date, result[day][0].name, result[day][0].colorNames)
                }
            }
```

**Reply by emagnier** [reply]
Because you are looking for the `periods`, not the object `key`.

```js
if (result[day][0].key === 'advent_january_3' && new Date(result[day][0].date).getFullYear() === 2022) {
  // Apply your fix
}
```

And not related to your point, but FYI: the `periods` metadata is an Array, that can contain data in different order. In your code above, you are just testing the first item in this array (with `[0]`). But what about if "DAYS_BEFORE_EPIPHANY" is in a second position? Instead, you should use the in [.includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) method to test if something is in the array.

**Reply by pvanthony** [reply]
Thank you very much for the example code. Saved me much time. Really did. Thank you again. Here is the code finally used to fix for year 2022.
```
        else if (result[day][0].key === 'christmas_time_january_3' && new Date(result[day][0].date).getFullYear() === 2022) {
            var calenderLink = url+ "CHRISTMASTIDE" + "/" + "WEEKDAY" + "/" + result[day][0].cycles.weekdayCycle + "/EPIPHANY/" + result[day][0].calendar.dayOfWeek
            console.log(calenderLink)
        }
```

Tried to do the fix for the other years but it looks much more complicated. Now I can understand how complex romcal is. You guys are really doing tough work.

Please note that we do appreciate all the work you guys are doing. Thank you for sharing this great work with the world.

**Reply by tukusejssirs** [reply]
@pvanthony, just to let you know, this issue is about to be fixed with PR #328.

**Reply by pvanthony** [reply]
Great!!!

**Reply by tukusejssirs** [reply]
Actually, now it is merged into the `dev` branch. :wink:

**Reply by pvanthony** [reply]
Have compiled it and it is on the test site now. Friar Julian will check through. If all ok, we will put it on the live site.
It is great that the fix was today as it is Friar Julian's birthday today. Great birthday present. :-)
